From 488e5cc608c3103a78a4af97a933463488a168a4 Mon Sep 17 00:00:00 2001
From: YoungJun <youngjun@telechips.com>
Date: Mon, 21 Sep 2020 01:28:49 +0000
Subject: [PATCH 2/3] waylandsink: support I420 format for dmabuf

- If gst video format is I420, shm memory for cpu rendering is selected.
- But GPU can make a texture for I420(YUV420)format, so add the
DRM_FORMAT_YUV420 format to cap.
---
 ext/wayland/wlvideoformat.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ext/wayland/wlvideoformat.c b/ext/wayland/wlvideoformat.c
index 77cfac8..8e54839 100644
--- a/ext/wayland/wlvideoformat.c
+++ b/ext/wayland/wlvideoformat.c
@@ -110,6 +110,7 @@ static const wl_DmabufVideoFormat dmabuf_formats[] = {
   {DRM_FORMAT_RGB565, GST_VIDEO_FORMAT_RGB16},
   {DRM_FORMAT_YUYV, GST_VIDEO_FORMAT_YUY2},
   {DRM_FORMAT_NV12, GST_VIDEO_FORMAT_NV12},
+  {DRM_FORMAT_YUV420, GST_VIDEO_FORMAT_I420},
 };
 
 enum wl_shm_format
-- 
2.24.1

